src/main/java/org/shanerx/tradeshop/listeners/ShopCreateListener.java

```diff
@@ -0,0 +1,148 @@
+
+	@SuppressWarnings("deprecation")
+	@EventHandler
+	public void onSignChange(SignChangeEvent event) {
+		//TODO add support for chestless itrade, 1 sign per chest
+
+		Sign shopSign = (Sign) event.getBlock().getState();
+		shopSign.setLine(0, event.getLine(0));
+		shopSign.setLine(1, event.getLine(1));
+		shopSign.setLine(2, event.getLine(2));
+		shopSign.setLine(3, event.getLine(3));
+
+		if (!ShopType.isShop(shopSign)) {
+			return;
+		}
+		ShopType shopType = ShopType.getType(shopSign);
+		Player p = event.getPlayer();
+		ShopUser owner = new ShopUser(p, ShopRole.OWNER);
+
+		if (!shopType.checkPerm(p)) {
+			failedSign(event, shopType, Message.NO_TS_CREATE_PERMISSION);
+		}
+
+		if (!checkShopChest(shopSign.getBlock()) && !shopType.isITrade()) {
+			failedSign(event, shopType, Message.NO_CHEST);
+			return;
+		}
+
+		ShopChest shopChest;
+		Shop shop;
+		Block chest = findShopChest(event.getBlock());
+
+		if (!shopType.isITrade()) {
+			if (ShopChest.isShopChest(chest)) {
+				shopChest = new ShopChest(chest.getLocation());
+			} else {
+				shopChest = new ShopChest(chest, p.getUniqueId(), shopSign.getLocation());
+			}
+
+			if (shopChest.hasOwner() && !shopChest.getOwner().equals(owner.getUUID())) {
+				failedSign(event, shopType, Message.NO_SHOP_PERMISSION);
+				return;
+			}
+
+			if (shopChest.hasShop() && !shopChest.getShopSign().getLocation().equals(shopSign.getLocation())) {
+				failedSign(event, shopType, Message.EXISTING_SHOP);
+				return;
+			}
+
+			shop = new Shop(new Tuple<>(shopSign.getLocation(), shopChest.getChest().getLocation()), shopType, owner);
+			shopChest.setName();
+
+
+			if (shopChest.isEmpty() && shop.hasProduct()) {
+				p.sendMessage(Message.EMPTY_TS_ON_SETUP.getPrefixed());
+			}
+		} else {
+			shop = new Shop(shopSign.getLocation(), shopType, owner);
+		}
+
+		ItemStack product = lineCheck(event.getLine(1)),
+				cost = lineCheck(event.getLine(2));
+
+		if (product != null)
+			shop.setProduct(product);
+
+		if (cost != null)
+			shop.setCost(cost);
+
+		if (shop.missingItems() && !shopType.isITrade()) {
+			event.setLine(0, ChatColor.GRAY + shopType.toHeader());
+		} else {
+			event.setLine(0, ChatColor.DARK_GREEN + shopType.toHeader());
+		}
+
+		event.setLine(3, shop.getStatus().getLine());
+		shop.saveShop();
+		shop.updateSign(event);
+
+		p.sendMessage(Message.SUCCESSFUL_SETUP.getPrefixed());
+	}
+
+	private ItemStack lineCheck(String line) {
+		if (line == null || line.equalsIgnoreCase("") || !line.contains(" ") || line.split(" ").length != 2)
+			return null;
+
+		String[] info = line.split(" ");
+
+		for (String str : info) {
+			if (str == null || str.equalsIgnoreCase(""))
+				return null;
+		}
+
+		if (!isInt(info[0]) || Material.matchMaterial(info[1]) == null)
+			return null;
+
+		int amount = Integer.parseInt(info[0]);
+		ItemStack product = new ItemStack(Material.matchMaterial(info[1]), amount);
+
+		if (isBlacklistItem(product))
+			return null;
+
+		return product;
+	}
+}
```

Comment:
Has anything changed here? I can't figure out what's new about this, in terms of functionality.

Reply:
First it got combined and stripped, then i added back the ability to type the items directly on the sign. Which is what lineCheck is doing. however if line[1] and line[2] are not typed correctly to be used they will be wiped and the shop will create without items.

Reply:
So I assume no mechanics were changed?

Reply:
The shop will create with line 1 & 2 empty and the status closed(which it will do whether or not items are present) and the player would have to use /addcost and /addproduct to set the items for the shop before opening it.